src/Components/Operations.js

```diff
@@ -0,0 +1,53 @@
+import React, {Component } from 'react';
+
+class Operations extends Component {
+constructor(){
+    super()
+    this.state={
+            amount: null,
+            vendor: '',
+            category: ''
+    }
+        
+}
+    addTrans=(e)=>{
+      let input = e.target.value== Number?+(e.target.value):e.target.value
+      let name = e.target.name   
+      this.setState({
+       [name] :input 
+      },function(){
+          console.log(this.state)
+      })
+      
+    }
+    withdraw=(e)=>{
+        let newstate=this.state
+         newstate.amount*= -1
```

Comment:
you are mutating the state directly, instead, you should make a deep copy using the three dots